packaging: make infernalis -> jewel upgrade work

smithfarm opened the pull request:
Attempts to install jewel ceph-common, ceph-mon, ceph-osd, and ceph-base
package over infernalis ceph package fail due to files existing in both.

See comment #4 in the tracker issue for a deeper analysis.

http://tracker.ceph.com/issues/15047 Fixes: #15047

Signed-off-by: Nathan Cutler <ncutler@suse.com>

smithfarm commented:
@ktdreyer @branto1 This is untested. Could look it over first? There's probably something obvious I'm missing.

smithfarm commented:
We could merge this PR into #7931 if that makes sense.

ktdreyer commented:
I think I see what you mean now. I'm ok with this, if it gets teuthology to pass. Are the RPM `Conflicts`/`Obsoletes` really necessary?

smithfarm commented:
> Are the RPM Conflicts/Obsoletes really necessary?

I will prepare a branch without the RPM commit and test that first.

smithfarm commented:
pushed wip-15047-deb-only to gitbuilders

branto1 commented on ceph.spec.in:
```diff
@@ -255,6 +255,8 @@ Requires:      lsb-release
 Requires:      redhat-lsb-core
 %endif
+Conflicts:      ceph < %{epoch}:10
+Obsoletes:      ceph < %{epoch}:10
```
I don't think we need the Conflicts line here (and in all the other cases). The Conflicts line would probably only cause problems for us.

As for the Obsoletes line that does make sense, not sure if it is actually necessary though as this might already be covered by the tight 'Requires: <exact_version>' lines.

branto1 commented on ceph.spec.in:
```diff
@@ -277,6 +279,8 @@ Requires(pre):	pwdutils
 Requires:	python-argparse
 %endif
+Conflicts:      ceph < %{epoch}:10
+Obsoletes:      ceph < %{epoch}:10
```
Hmm, did we put any of the files from the ceph package to ceph-common in the package split?

smithfarm replied:
@branto1: Yes, `ceph_daemon.py` got moved in the package split - see http://tracker.ceph.com/issues/15046 - but this bug only affected Debian, so I will remove these lines.

smithfarm commented:
Repushed the RPM patch with the following changes:

1. removed all the `Conflicts:` lines
2. removed `Obsoletes:` line from ceph-common
3. added `Obsoletes:` line to ceph-mds

smithfarm commented:
Scheduled a full `upgrade/infernalis-x` suite on the deb-only integration branch:

```
smithfarm@teuthology:~/teuthology$ ./virtualenv/bin/teuthology-suite --verbose \
--priority 999 --suite upgrade/infernalis-x --suite-branch jewel --email ncutler@suse.cz \
--ceph wip-15047-deb-only --machine-type vps machine_types/vps.yaml
```

http://pulpito.ceph.com/smithfarm-2016-03-11_01:41:18-upgrade:infernalis-x-wip-15047-deb-only---basic-vps/

smithfarm commented:
@yuriw Can you check the teuthology command line in the previous comment? Hopefully I got it right . . .

smithfarm commented:
Test completed, logs of failed and dead jobs analyzed. Conclusions:

1. the RPM commit is not needed because the issue does not manifest in RPM
2. the DEB commit fixes the issue

ktdreyer commented:
Thanks @smithfarm . We can merge the debian commit, then.

smithfarm commented:
RPM commit dropped. Rebased on latest jewel.

ktdreyer commented:
thanks for figuring this out!
